src/impl/tpls/KokkosBlas_gesv_tpl_spec_decl.hpp

```diff
@@ -100,11 +100,11 @@ struct GESV< \
     \
     int info = 0; \
     \
-    if(pivot_t) { \
-      int *ipiv = NULL; \
-      ipiv = (int*)malloc(N*sizeof(int)); \
+    if(with_pivot) { \
+      int *ipiv = nullptr; \
+      ipiv = new int[N]; \
```

Comment:
Why doesn't the function just take an array of pivots, that is unused (and could be null) if `with_pivot` is false?

Reply:
What @kyungjoo-kim  said, we even don't need ```with_pivot```. Users just need to provide zero-extent pivot view for no pivot case.

Reply:
or should we use ```with_pivot``` (false) and zero-extent pivot view?

Reply:
It is just my preference. I do not like putting duplicated information (zero length array and with_pivot arguments can contain common information as ``with_pivot = ipiv.extent(0) != 0``). @mhoemmen comment is that it can cause unused variable warning but as we check its length the variable is always used for both cases.

Reply:
`with_pivot` is a run-time option, so as long as you use the input pivot array inside that case, there should be no variable warning.

You should be able to dispatch straight to LAPACK in the host case.  `DGESV` [takes a pivot array](http://www.netlib.org/lapack/explore-html/d7/d3b/group__double_g_esolve_ga5ee879032a8365897c3ba91e3dc8d512.html).